File: leetcode/014_longestcommonprefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        
        strs = sorted(strs)
        count = 0
        
        str1 = strs[0]
        str2 = strs[len(strs) - 1]
        maxLen = min(len(str1), len(str2))

        while count < maxLen:
            if str1[count] != str2[count]:
                break
            count += 1
        
        return str1[0:count]

File: leetcode/014_longestcommonprefix.ts
function longestCommonPrefix(strs: string[]): string {
    strs.sort()
    const firstWord = strs[0]
    const lastWord = strs[strs.length - 1]
    const maxLen = Math.min(firstWord.length, lastWord.length)
    let idx = 0
    for (let i = 0; i < maxLen; i++) {
        if (firstWord[i] !== lastWord[i]) break
        idx += 1
    }

    return firstWord.substring(0, idx)
};
